### ISSUE TITLE
Multilabel classification (more than one label)

### ISSUE THREAD

**Issue by tagwato** [huggingface/setfit#337]
Hi,
I tried to map more than one column/feature as "label" in the column_mapping but it was not accepted, even when the multi_target_strategy attribute was set.
Is there a way to handle a label that represents a multidimensional value/a vector?
Tks.

**Comment by tomaarsen** [comment]
Yes, that is possible. Consider this example script:
```python
from datasets import Dataset
import evaluate

from setfit import SetFitModel, SetFitTrainer

model = SetFitModel.from_pretrained(
    "sentence-transformers/paraphrase-mpnet-base-v2",
    multi_target_strategy="one-vs-rest",
)
dataset = Dataset.from_dict({"text_new": ["", "a", "b", "ab"], "label_new": [[0, 0], [1, 0], [0, 1], [1, 1]]})

multilabel_f1_metric = evaluate.load("f1", "multilabel")
multilabel_accuracy_metric = evaluate.load("accuracy", "multilabel")


def compute_metrics(y_pred, y_test):
    return {
        "f1": multilabel_f1_metric.compute(predictions=y_pred, references=y_test, average="micro")["f1"],
        "accuracy": multilabel_accuracy_metric.compute(predictions=y_pred, references=y_test)["accuracy"],
    }


trainer = SetFitTrainer(
    model=model,
    train_dataset=dataset,
    eval_dataset=dataset,
    metric=compute_metrics,
    num_iterations=1,
    column_mapping={"text_new": "text", "label_new": "label"},
)

trainer.train()
metrics = trainer.evaluate()

print(metrics)
breakpoint()
```
or [this notebook](https://github.com/huggingface/setfit/blob/main/notebooks/multilabel_HoC.ipynb).
In short, for an N-class multiclass scenario, each label must be a list of N elements, where each element is either a 1 or a 0, depending on whether that class is true or false for the given sample.

If your labels are in the format of `[2, 5]` to indicate that classes with indices 2 and 5 are True, while all other classes are False, then you can use the [`MultiLabelBinarizer`](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MultiLabelBinarizer.html) from sklearn to help you out:
```python
from sklearn.preprocessing import MultiLabelBinarizer
import numpy as np

# Your annotated labels
labels = [[5, 3], [2, 3], [1, 0, 7], [6, 4, 5]]

# Let's binarize the labels
binarizer = MultiLabelBinarizer()
binarized_labels = binarizer.fit_transform(labels)
print(binarized_labels)
# =>
# [[0 0 0 1 0 1 0 0]
#  [0 0 1 1 0 0 0 0]
#  [1 1 0 0 0 0 0 1]
#  [0 0 0 0 1 1 1 0]]

# Now we can use them in our Dataset and train a model.
# ...

# Once trained, we'll get preditions in the same binarized style again.
# We can revert back to our original label format like so:
# (For example purposes I'll set the predictions manually)
predictions = np.array([[0, 0, 1, 0, 0, 1, 0, 0], [0, 1, 0, 1, 0, 1, 0, 0]])
non_binarized_predictions = binarizer.inverse_transform(predictions)

print(non_binarized_predictions)
# =>
# [(2, 5), (1, 3, 5)]
```

I hope that helps.

- Tom Aarsen

**Comment by tagwato** [comment]
Thank you very much @tomaarsen !

I mistakenly thought that setfit would somehow do the hot-encoding.  
Now I understand: we need to define that transformation manually.

**Comment by tomaarsen** [comment]
Exactly! Perhaps in the future, SetFit can detect this and perform this transformation automatically. 
For now, I'll close this issue. Feel free to reopen or make a new issue if you have further problems.

- Tom Aarsen

**Comment by aaronvg** [comment]
How does one get the probabilities out of the prediction?
